components/camera/videosource/helpers.go

```diff
@@ -44,34 +63,29 @@ func readyBytesFromURL(ctx context.Context, client http.Client, url string) ([]b
 	return io.ReadAll(body)
 }
 
-func checkLazyFromData(ctx context.Context, data []byte) (image.Image, bool, error) {
-	requestedMime := gostream.MIMETypeHint(ctx, "")
-	if actualMime, isLazy := utils.CheckLazyMIMEType(requestedMime); isLazy {
-		usedMimeType := http.DetectContentType(data)
-		if actualMime != "" && actualMime != usedMimeType {
-			return nil, false,
-				errors.Errorf("mime type requested (%q) for lazy decode not returned (got %q)",
-					actualMime, usedMimeType)
-		}
-
-		return rimage.NewLazyEncodedImage(data, usedMimeType, -1, -1), true, nil
-	}
-	return nil, false, nil
-}
-
 func readColorURL(ctx context.Context, client http.Client, url string) (image.Image, error) {
 	colorData, err := readyBytesFromURL(ctx, client, url)
 	if err != nil {
 		return nil, errors.Wrap(err, "couldn't ready color url")
 	}
 
-	if lazyImg, ok, err := checkLazyFromData(ctx, colorData); err != nil {
-		return nil, err
-	} else if ok {
-		return lazyImg, nil
+	requestedMime := gostream.MIMETypeHint(ctx, "")
+	actualMime, isLazy := utils.CheckLazyMIMEType(requestedMime)
+	detectedMimeType := http.DetectContentType(colorData)
+	if isLazy {
+		if actualMime != "" && actualMime != detectedMimeType {
+			return nil, errors.Errorf(
+				"mime type requested (%q) for lazy decode not returned (got %q)",
+				actualMime, detectedMimeType,
+			)
+		}
+		return rimage.NewLazyEncodedImage(colorData, detectedMimeType), nil
+	}
+	if actualMime == "" {
+		actualMime = detectedMimeType
 	}
 
-	img, err := decodeImage(colorData)
+	img, err := rimage.DecodeImage(ctx, colorData, actualMime)
```

Comment:
At this point, we still don't know that actualMime==detectedMimeType, so we could be passing the wrong mime type to DecodeImage().

Reply:
is this a current concern? right now, we're using image.Decode and that deduces the mime type from the data, not one supplied by us. There are multiple scenarios right now where requestedMime in the existing checkLazyFromData function turns out to be an empty string and the read urls functions don't care and let image.Decode handle it. Should I change the functions to not accept this scenario?

Reply:
as a side note, I am changing the code and adding comments to at least make that situation clearer for now

Reply:
I didn't quite follow. What I meant was I think we should pass detectedMimeType to DecodeImage here instead of actualMime, since actualMime is the requested mime type and may not be what was returned.